### PR TITLE
Create `Attestation` logic in `StakingRewards`

### PR DESCRIPTION
**Pull request by 0xfuturistic**

### CODE REVIEW

File: src/example-stakers/CrossDomainStaker/L2/StakingRewards.sol [hunk 1]
```diff
@@ -3,161 +3,249 @@ pragma solidity ^0.8.23;
...
-    }
+        // ensure the block hash is valid
+        if (_blockHash != blockhash(_blockNumber)) revert InvalidBlockHash();
 
-    /// @notice Distributes `_value` amount of reward tokens to all stakers.
-    /// @param _value The value amount of reward tokens to distribute.
-    function _distributeReward(uint256 _value) internal {
-        if (totalSupply == 0) revert NoStakes();
+        // use epoch start balance as value for the attestation
+        _value = attesterData.lastSyncedEpochNumber == currentEpochNumber
```
Comment (josojo) [inline src/example-stakers/CrossDomainStaker/L2/StakingRewards.sol:230]: maybe value should be called balance?

File: src/example-stakers/CrossDomainStaker/L2/StakingRewards.sol [hunk 2]
```diff
@@ -3,161 +3,249 @@ pragma solidity ^0.8.23;
...
-    }
+        // update attester
+        attesterData.lastAttestedEpochNumber = currentEpochNumber;
 
-    /// @notice Accrues rewards for `_beneficiary`.
-    /// @param _beneficiary The address of the beneficiary.
-    /// @return _value The amount of reward tokens accrued.
-    function _accrueRewards(address _beneficiary) internal returns (uint256 _value) {
-        _value = stakes[_beneficiary].mulWad(rewardPerToken - lastRewardPerToken[_beneficiary]);
+        (bool success,) = _attester.call{value: _reward}("");
```
Comment (josojo) [inline src/example-stakers/CrossDomainStaker/L2/StakingRewards.sol:242]: We could do this only if rewards >0

File: src/MultiStaker.sol [hunk 3]
```diff
@@ -208,25 +220,33 @@ contract MultiStaker {
...
             _canSlash: _canSlash,
             _owner: msg.sender,
-            _value: deposit_.value,
             _unstakeGasAmount: _unstakeGasAmount,
             _extraData: _extraData
         });
 
-        if (STAKE_TOKEN.balanceOf(address(this)) != expectedBalance1) revert TransferFailed();
+        // validate that the right amount was transferred
+        if (STAKE_TOKEN.balanceOf(address(this)) != balance0 - (isPossessory_ ? deposit_.value : 0)) {
```
Comment (josojo) [inline src/MultiStaker.sol:242]: I think it was easier to read with expectedBalance... but up to u

**Review by josojo — APPROVED** [review]
nice

File: src/example-stakers/CrossDomainStaker/L2/StakingRewards.sol (hunk 2 above)
Comment (0xfuturistic) [reply src/example-stakers/CrossDomainStaker/L2/StakingRewards.sol:242]: good idea!

**Pull request merged**